src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -573,9 +573,30 @@ uintptr_t CoffNativeCodeManager::GetConservativeUpperBoundForOutgoingArgs(Method
 
         upperBound = dac_cast<TADDR>(context.Sp);
 #else
-        PORTABILITY_ASSERT("GetConservativeUpperBoundForOutgoingArgs");
-        upperBound = NULL;
-        RhFailFast();
+        PTR_uint8_t gcInfo;
+        uint32_t codeOffset = GetCodeOffset(pMethodInfo, (PTR_VOID)pRegisterSet->IP, &gcInfo);
+
+        hdrInfo infoBuf;
+        size_t infoSize = DecodeGCHdrInfo(GCInfoToken(gcInfo), codeOffset, &infoBuf);
+        PTR_CBYTE table = gcInfo + infoSize;
+
+        REGDISPLAY registerSet = *pRegisterSet;
+
+        if (::UnwindStackFrameX86(&registerSet,
```

Comment:
When is this expected to fail?

Reply:
It’s not, I can change it to ASSERT.

Reply:
Actually, I checked, with funclets enabled it is guaranteed to return `true`.